lib/ansible/plugins/cache/__init__.py

```diff
@@ -207,10 +207,20 @@ def has_expired(self, key):
         return True
 
     def keys(self):
+        # When using a prefix we must remove it from the key name before
+        # checking the expiry and returning it to the caller. Keys that do not
+        # share the same prefix cannot be fetched from the cache.
+        prefix = self.get_option('_prefix')
```

Comment:
Sorry for missing your update. The tests look good, thanks for including those. I just noticed that the prefix can be None, which will fail with `len`.
```suggestion
        prefix = self.get_option('_prefix') or ''
```

Reply:
instead of `or ''` just set the default to `''`, you should never need conditionals on `get_option`, otherwise the docs wont really reflect the real options

Reply:
also might need to add aliases .. _prefix should be prefix since it is not hidden/internal/unsettable option (my mistake from initial conversion of the plugins, kept those names)

Reply:
another random thought ... we need to specify 'required common options' per plugin type, in this case 'prefix' is required for cache plugins.

Reply:
No, both inventory and fact cache have non-null defaults for prefix already. It occured to me someone might do:
```
# ansible.cfg
[inventory]
cache_plugin=jsonfile
caching_connection=/tmp/mycache
caching_prefix=
```
And then the only other place we care about prefix handles this gracefully already:
```
    def _get_cache_file_name(self, key):
        prefix = self.get_option('_prefix')
        if prefix:
            cachefile = "%s/%s%s" % (self._cache_dir, prefix, key)
        else:
            cachefile = "%s/%s" % (self._cache_dir, key)
        return cachefile
```
Good point about aliases, I can update as a follow-up PR.

Reply:
It looks like it's only an empty string though in that case though, so I rescind that